File: day06.py
with open('day06_input.txt') as f:
    buffer = list(f.read())


def check_marker(marker):
    for i in range(len(marker)):
        rest_of_marker = marker[i+1: len(marker)+1]
        if (marker[i] in rest_of_marker):
            return False
    return True


def check_buffer(buff):
    for i in range(4, len(buffer)):
        marker = buffer[i-4:i]
        if check_marker(marker):
            return i


print(check_buffer(buffer))

File: day06.js
const buffer = require("fs")
  .readFileSync("day06_input.txt")
  .toString()
  .split("")

let answer = checkBuffer(buffer)

function checkBuffer(buffer) {
  for (let i = 4; i < buffer.length; i++) {
    let marker = buffer.slice(i - 4, i)
    console.log(marker)
    if (isUniqueMarker(marker)) {
      return i
    }
  }
}

console.log(answer)

function isUniqueMarker(marker) {
  for (let j = 0; j < marker.length; j++) {
    let restOfMarker = marker.slice(j + 1, marker.length + 1)
    //console.log(restOfMarker)
    if (restOfMarker.includes(marker[j])) {
      return false
    }
  }
  return true
}
